shell/components/nav/NotificationCenter/Notification.vue

```diff
@@ -76,9 +78,17 @@ const age = computed(() => {
 
 const clz = computed(() => CLASSES[props.item.level]);
 
-// Open a URL from either the primary or secondary buttons in a new tab
+// Invoke action on either the primary or secondary buttons
+// This can open a URL in a new tab OR navigate to an application route
 const action = (action: NotificationAction) => {
-  window.open(action.target, '_blank');
+  if (action.target) {
+    window.open(action.target, '_blank');
+  } else if (action.route) {
+    router.push(action.route);
```

Comment:
Also this one should be fine. Maybe you could test first if the route is good or add a catch, because you never know.